packages/gui/src/components/chiatools/logViewer/LogViewer.tsx

```diff
@@ -0,0 +1,562 @@
+import { Flex, useShowError } from '@chia-network/core';
+import { Trans } from '@lingui/macro';
+import { Box, CircularProgress, Typography, TextField, Button } from '@mui/material';
+import { useTheme } from '@mui/material/styles';
+import React, { useEffect, useState, useCallback } from 'react';
+
+import decodeError from '../../../electron/utils/decodeError';
+
+import LogViewerToolbar from './LogViewerToolbar';
+import {
+  LogLevel,
+  LogViewerFilter,
+  LogViewerProps,
+  LogFileInfo,
+  CachedContent,
+  PaginationInfo,
+} from './LogViewerTypes';
+import { LOG_COLORS, CACHE_TIMEOUT, DEFAULT_FILTER, filterLogContent } from './LogViewerUtils';
+
+const highlightSearchText = (text: string, searchTerm: string | undefined) => {
+  if (!searchTerm) return text;
+
+  const parts = text.split(new RegExp(`(${searchTerm})`, 'gi'));
+  return parts.map((part, index) =>
+    part.toLowerCase() === searchTerm.toLowerCase() ? (
+      <span
+        // eslint-disable-next-line react/no-array-index-key -- Using index is safe here as the array is stable and order won't change (requested by ChiaMineJP)
+        key={`highlight-${part}-${index}`}
+        style={{
+          backgroundColor: 'rgba(255, 255, 0, 0.3)',
+          fontWeight: 'bold',
+        }}
+      >
+        {part}
+      </span>
+    ) : (
+      part
+    ),
+  );
+};
+
+declare global {
+  interface Window {
+    chiaLogs: {
+      getContent: () => Promise<{
+        content?: string;
+        path?: string;
+        size?: number;
+        error?: string;
+      }>;
+      getInfo: () => Promise<{
+        path: string;
+        exists: boolean;
+        size: number;
+        readable: boolean;
+        error?: string;
```

Comment:
- `error?: string` should be `fileError?: string;`
- `defaultPath?: string;` is missing here.